src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -154,6 +155,22 @@ void QgsMeshLayer::setRendererSettings( const QgsMeshRendererSettings &settings
   triggerRepaint();
 }
 
+QgsMesh3dAveragingMethod *QgsMeshLayer::averagingMethod() const
+{
+  return mAveragingMethod.get();
+}
+
+void QgsMeshLayer::setAveragingMethod( QgsMesh3dAveragingMethod *method )
+{
+  bool needsRepaint = QgsMesh3dAveragingMethod::equals( method, mAveragingMethod.get() );
+  if ( needsRepaint )
+  {
+    mAveragingMethod.reset( method );
+    emit rendererChanged();
+    triggerRepaint();
+  }
+}
```

Comment:
We probably don't need averaging method directly attached to mesh layer - it could be taken out of rendering settings.